springwolf-plugins/springwolf-spring-cloud-rabbit-binder-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/CloudStreamRabbitBinderChannelsScanner.java

```diff
@@ -0,0 +1,198 @@
+            return Maps.immutableEntry(channelName, channelItem);
+        }
+        return null;
+    }
+
+    private ChannelItem buildChannel(Map<String, ? extends ChannelBinding> channelBinding,
+                                     Class<?> payloadType,
+                                     Map<String, ? extends OperationBinding> operationBinding) {
+        String modelName = schemasService.register(payloadType);
+
+        Message message = Message.builder()
+                .name(payloadType.getName())
+                .title(modelName)
+                .payload(PayloadReference.fromModelName(modelName))
+                .build();
+
+        Operation operation = Operation.builder()
+                .message(message)
+                .bindings(operationBinding)
+                .build();
+
+        return ChannelItem.builder()
+                .bindings(channelBinding)
+                .publish(operation)
+                .build();
+    }
+
+    protected Map<String, ? extends OperationBinding> buildOperationBinding(String channelName) {
+        List<String> routingKeys = new ArrayList<>();
+        routingKeys.add(this.bindingsMap.get(channelName).getRoutingKey());
+        return ImmutableMap.of("amqp", AMQPOperationBinding.builder()
+                .cc(routingKeys)
+                .build());
+    }
+
+    private Optional<String> getChannelNameForMethod(String methodName) {
+        return this.bindingsMap.keySet().stream().filter(channelName -> channelName.startsWith(methodName+"-in-")).findFirst();
+    }
+
+    protected Map<String, ? extends ChannelBinding> buildChannelBinding(String channelName) {
+        AMQPChannelBinding.ExchangeProperties exchangeProperties = new AMQPChannelBinding.ExchangeProperties();
+        AMQPChannelBinding.QueueProperties queueProperties = new AMQPChannelBinding.QueueProperties();
+        Binding binding = bindingsMap.get(channelName);
+        exchangeProperties.setName(binding.getExchange());
+        queueProperties.setName(binding.getDestination());
+        return ImmutableMap.of("amqp", AMQPChannelBinding.builder()
+                .is("routingKey")
+                .exchange(exchangeProperties)
+                .queue(queueProperties)
+                .build());
+    }
+
+    private Class<?> getReturnTypeOfMethod(Method method) {
+        Type genericReturnType = method.getGenericReturnType();
+        if(genericReturnType != null && genericReturnType instanceof ParameterizedTypeImpl) {
+            Type[] typeArguments = ((ParameterizedTypeImpl) genericReturnType).getActualTypeArguments();
+            if(typeArguments != null && typeArguments.length == 1) {
+                String typeName = typeArguments[0].getTypeName();
+                try {
+                    return Class.forName(typeName);
+                } catch(ClassNotFoundException classNotFoundException) {
+                    //should never happen;
+                }
+            }
+        }
+        return null;
```

Comment:
Is this ok to return `null` here? If this is unexpected, we should log warn and and not map this method to a channel I think